src/context/WalletProvider/NativeWallet/NativePasswordRequired/hooks/useStateToggle/useStateToggle.ts

```diff
@@ -0,0 +1,7 @@
+import { useState } from 'react'
+
+export const useStateToggle = (): [boolean, () => void] => {
+  const [state, setState] = useState<boolean>(false)
+  const toggleState = () => setState(s => !s)
+  return [state, toggleState]
```

Comment:
This looks exactly like `useDisclosure` with `{ isOpen, onToggle }`. This is probably fine though.

Reply:
Yea I was not sure if it made since to use useDisclosure because it is very open close oriented in it's naming conventions. And in this use case it was used for hiding and showing not opening and closing. That is why I ended up doing that but I could see how it maybe better to useDisclosure and just change the variable names to something that is not open/close oriented.

Reply:
I decided to just go with using useDisclosure and renaming the variables so it aligned with the functionality.